commands/registerHandlers.go

```diff
@@ -0,0 +1,188 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Strum355/log"
+	"github.com/UCCNetsoc/discord-bot/api"
+	"github.com/UCCNetsoc/discord-bot/prometheus"
+	"github.com/bwmarrin/discordgo"
+	"github.com/dghubble/oauth1"
+	twitterApi "github.com/ericm/go-twitter/twitter"
+	"github.com/spf13/viper"
+)
+
+var (
+	// Twitter
+	twitterClient *twitterApi.Client
+)
+
+// Reaction on a user message
+type Reaction string
+
+const (
+	twitter Reaction = "🇹"
+)
+
+var (
+	commandsMap = make(map[string]func(context.Context, *discordgo.Session, *discordgo.InteractionCreate))
+	reactionMap = make(map[string]interface{}) // Maps message ids to content
+)
+
+type commandFunc func(context.Context, *discordgo.Session, *discordgo.InteractionCreate)
+
+func command(name string, function commandFunc) {
+	commandsMap[name] = function
+}
+
+// Register command handlers
+func RegisterHandlers(s *discordgo.Session) {
+	// TODO: Clean up repetition in registering commands and registering command handlers (e.g declaring command names both here and in the registerCommands.go)
+	RegisterCommands(s)
+	// Public commands
+	command("ping", ping)
+	command("version", version)
+	command("members", members)
+	command("dig", dig)
+	command("corona", coronaCommand)
+	command("vaccines", vaccines)
+	command("boosters", boostersCommand)
+	command("upcoming", upcomingEvent)
+	command("online", online)
+	// Committee commands
+	command("up", checkUpCommand)
+	command("shorten", shortenCommand)
+
+	// Setup APIs
+	twitterConfig := oauth1.NewConfig(viper.GetString("twitter.key"), viper.GetString("twitter.secret"))
+	twitterToken := oauth1.NewToken(viper.GetString("twitter.access.key"), viper.GetString("twitter.access.secret"))
+	httpClient := twitterConfig.Client(oauth1.NoContext, twitterToken)
+	twitterClient = twitterApi.NewClient(httpClient)
+
+	// Setup Interaction Handlers
+	s.AddHandler(func(s *discordgo.Session, i *discordgo.InteractionCreate) {
+		callCommand(s, i)
+	})
+
+	// Setup Message Handlers
+	s.AddHandler(messageCreate)
+	s.AddHandler(messageReaction)
+	s.AddHandler(messageDelete)
+	s.AddHandler(serverJoin)
```

Comment:
We can deleted the handlers that won't work with the new API.

Reply:
I'm pretty sure since we are just getting all non privileged intents by default these wont actually break/get faded out of the api. I can remove them though if we are sure on it

Reply:
The message react was only needed for the old events method and server join only works with privileges so we should delete those anyway.